test/powershell/Modules/Microsoft.PowerShell.Core/PSSessionConfiguration.Tests.ps1

```diff
@@ -0,0 +1,848 @@
+                        $Result.Error | Should be $null
+                    }
+
+                    # Create Test Startup Script
+                    function CreateStartupScript {
+                        $ScriptContent = @"
+`$global:testvariable = "testValue"
+"@
+                        
+                        $TestScript = join-path $global:TestDir "StartupTestScript.ps1"
+                        $null = Set-Content -path $TestScript -Value $ScriptContent
+                    
+                        return $TestScript
+                    }
+
+                    # Create new Config File
+                    function CreateTestConfigFile {
+                    
+                        $TestConfigFile = join-path $global:TestDir "TestConfigFile.pssc"
+                        $null = New-PSSessionConfigurationFile -Path $TestConfigFile -SessionType Default
+                        return $TestConfigFile
+                    }
+
+                    function CreateTestModule {
+                        $ScriptContent = @"
+function IsTestModuleImported {
+return `$true
+}
+Export-ModuleMember IsTestModuleImported
+"@
+                        $TestModuleFileLoc = $global:TestDir
+
+                        if(-not (Test-path $TestModuleFileLoc))
+                        {
+                            $null = New-Item -Path $TestModuleFileLoc -ItemType Directory -Force -ErrorAction Stop
+                        }
+
+                        $TestModuleFile = join-path $TestModuleFileLoc "TestModule.psm1"
+                        $null = Set-Content -path $TestModuleFile -Value $ScriptContent
+                    
+                        return $TestModuleFile
+                    }
+
+                    function CreateTestAssembly {
+                        $PscConfigDef = @"
+using System;
+using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+using System.Management.Automation.Remoting;
+
+namespace PowershellTestConfigNamespace
+{
+    public sealed class PowershellTestConfig : PSSessionConfiguration
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="senderInfo"></param>
+        /// <returns></returns>
+        public override InitialSessionState GetInitialSessionState(PSSenderInfo senderInfo)
+        {
+            return InitialSessionState.CreateDefault();
+        }
+        
+    }
+}
+"@
+                        $global:Sma = [reflection.assembly]::load('System.Management.Automation, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL').location
+                        $global:SourceFile = join-path $global:TestDir "PowershellTestConfig.cs"
+                        $PscConfigDef | out-file $global:SourceFile -Encoding ascii -Force
+                        $TestAssemblyName = "TestAssembly.dll"
+                        $TestAssemblyPath = join-path $global:TestDir $TestAssemblyName
```

Comment:
`$TestAssemblyPath` is in the `$TestDrive` directory. Please don't use build an assembly in `$TestDrive`, because the assembly file cannot be removed once it's loaded, but Pester will try to remove `$TestDrive` and thus cause errors to be thrown even though the test passes. The error will make the test log hard to look at.
When generating an assembly using `Add-Type`, it's recommended to use a path in `[System.IO.Path]::GetTempPath()`